code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -1,101 +1,141 @@
 ////////////////INTERNAL MAGAZINES//////////////////////
 
+/obj/item/ammo_box/magazine/internal
+	desc = "Oh god, this shouldn't be here"
+
 //internals magazines are accessible, so replace spent ammo if full when trying to put a live one in
 /obj/item/ammo_box/magazine/internal/give_round(obj/item/ammo_casing/R)
 	return ..(R,1)
 
+
+
+// Revolver internal mags
 /obj/item/ammo_box/magazine/internal/cylinder
 	name = "revolver cylinder"
-	desc = "Oh god, this shouldn't be here"
 	ammo_type = /obj/item/ammo_casing/a357
 	caliber = "357"
 	max_ammo = 7
 
 /obj/item/ammo_box/magazine/internal/cylinder/ammo_count(countempties = 1)
-	if (!countempties)
-		var/boolets = 0
-		for (var/i = 1, i <= stored_ammo.len, i++)
-			var/obj/item/ammo_casing/bullet = stored_ammo[i]
-			if (bullet.BB)
-				boolets++
-		return boolets
-	else
-		return ..()
+	var/boolets = 0
+	for (var/i = 1, i <= stored_ammo.len, i++)
+		var/obj/item/ammo_casing/bullet = stored_ammo[i]
+		if(bullet && (bullet.BB || countempties))
+			boolets++
 
-/obj/item/ammo_box/magazine/internal/cylinder/rus357
-	name = "russian revolver cylinder"
-	desc = "Oh god, this shouldn't be here"
-	ammo_type = /obj/item/ammo_casing/a357
-	caliber = "357"
-	max_ammo = 6
-	multiload = 0
+	return boolets
 
-/obj/item/ammo_box/magazine/internal/cylinder/rus357/New()
-	stored_ammo += new ammo_type(src)
+/obj/item/ammo_box/magazine/internal/cylinder/get_round(keep = 0)
+	rotate()
+
+	var/b = stored_ammo[1]
+	if(!keep)
+		stored_ammo[1] = null
+
+	return b
+
+/obj/item/ammo_box/magazine/internal/cylinder/proc/rotate()
+	var/b = stored_ammo[1]
+	stored_ammo.Cut(1,2)
+	stored_ammo.Insert(0, b)
+
+/obj/item/ammo_box/magazine/internal/cylinder/proc/spin()
+	for(var/i = 1, i <= rand(0, max_ammo*2), i++)
+		rotate()
+
+
+/obj/item/ammo_box/magazine/internal/cylinder/give_round(obj/item/ammo_casing/R, replace_spent = 0)
+	if(!R || (caliber && R.caliber != caliber) || (!caliber && R.type != ammo_type))
+		return 0
+
+	for(var/i = 1, i <= stored_ammo.len, i++)

```

Comment:
I mean, you only have one genuine use of ```i```, and that's for putting ```R``` at the index of the bullet, which you can use ```find(bullet)``` to get it's index, however that's no faster.

Ultimately then, the fastest here is
```for(var/i in 1 to stored_ammo.len)```